jax/_src/lax/lax.py

```diff
@@ -1083,7 +1087,7 @@ def _tri(dtype: DType, shape: Shape, offset: int) -> Array:
                 broadcasted_iota(np.int32, shape, 1))
   return convert_element_type_p.bind(bool_tri, new_dtype=dtype, weak_type=False)
 
-def stop_gradient(x):
+def stop_gradient(x: T) -> T:
```

Comment:
Same here.

Reply:
It seems to me that the caller can force the type in the rare case they're pushing scalars in:
```
x: RealNumeric = stop_gradient(2.0)
```
Also, pushing concrete (non-tracer) scalars into to `stop_gradient` doesn't really make sense.

Reply:
On further inspection, it appears that `stop_gradient` seems to return concrete values unchanged, so this annotation should be fine?